agent_shared/src/error.rs

```diff
@@ -2,8 +2,13 @@ use thiserror::Error;
 
 #[derive(Error, Debug)]
 pub enum SharedError {
-    #[error("Error: {0}")]
-    Generic(String),
+    #[error("Error while loading `{0}`: {1}")]
+    GenericConfigurationError(String, String),
+    // This error should always be unreachable since all configuration fields should have a default value in development mode
+    #[error("Configuration parameter `{0}` is missing a default value")]
+    MissingDefaultValueForDevelopment(String),
+    #[error("Configuration is not suitable for development: {0}")]
+    ConfigurationNotSuitableForDevelopment(String),
```

Comment:
Any valid config will always be accepted in the Development profile, so this error shouldn't exist. Instead, we should have one generic `InvalidConfiguration` error which is thrown if an invalid combination of config values is set (such as `did:iota` and `in_memory`).